src/main/java/ui/components/NavigableListView.java

```diff
@@ -103,7 +103,7 @@ public void restoreSelection() {
             // Do not trigger event; selection did not conceptually change
         } else {
             // The item disappeared
-            if (getItems().size() == 0) {
+            if (getItems().size() == 0 || !selectedIndex.isPresent()) {
```

Comment:
Rather than do this, which is a bit odd given the body (if index is empty, set index to empty), we can move this to the `else` branch instead.

The comments here can also be improved, to say what we are doing.
```java
// Figure out what is selected based on previous and current state
if (itemFound) {
    // Maintain selection
} else {
    if (getItems().isEmpty()) {
        // Nothing can be selected
    } else if (selectedIndex.isPresent()) {
        // Consider what is currently selected to be the new selection
    }
}
```